### ISSUE TITLE
Easy and efficent way using / unlocking a password-and-key secured database

### ISSUE THREAD

**Issue by noMICROSOFTbuhtz** [keepassxreboot/keepassxc#2682]
## Summary
I use KeePassXC 2.3.4 on Debian stable with a database secured by password **and** key-file on a usb-stick. The current workflow looks like this

1. Start the PC, Browser (Firefox with KeePassXC-Browser AddOn) and go to a login page (e.g. [Codeberg.org](https://codeberg.org/user/login))
2. KeePassXC is started while booting but the database is not open yet.
3. Plug in the USB-Stick, it is automounted by my filemanager and KeePassXC recoginize it well
4. Click on KeePassXC systray-symbol enter the password
5. Now I am able to use KeePassXC.

## Desired Behavior
Point 3 & 4 could be merged. When I plug in the USB-Stick and KeePassXC recognize a key-file it should popup **itself** and ask for the database password.
Then is no need to click around. Just one move of my hand to put in the USB-Stick and some finger-tips for the password.

## Possible Solution
I am unsure if this behavior would have security issues? Automatics like this should always be optional and off by default.

## Context
It would ease the way to use KeePassXC. I always imagine how my mother would use something like KeePassXC. ;) It is very near to be useable by her! ;) Also I think using a stick as a "key" is more natural in the thinking/mind of none-IT-persons.

**Comment by RealOrangeOne** [comment]
Keepass doesn't care what's used as a keyfile, it's just data. It could possibly detect it by file extension. But there's no way of detecting if the keyfile is valid for the current vault, or which of the open vaults it's for, if any.

**Comment by phoerious** [comment]
There is a legacy key file format, which is detectable, but we deprecated it for various reasons. You also don't want KeePassXC to automatically scan and interpret files on plugged-in drives. What you really want is a YubiKey instead of a generic key file. You do not need to mount a YubiKey and KeePassXC automatically recognises one if it is plugged into your computer.

**Comment by noMICROSOFTbuhtz** [comment]
I don't understand your explanation.
KP does recognize the key file well. The key-file field is empty when the usb-stick is not mounted. When it is mounted the key-file field has the correct path to the file. I see no problem here.
